modules/core/kotlin/com/github/timaxinc/kcipt/classloader/BlacklistClassloader.kt

```diff
@@ -0,0 +1,141 @@
+package com.github.timaxinc.kcipt.classloader
+
+import com.github.timaxinc.kcipt.util.io.Block
+import com.github.timaxinc.kcipt.util.io.startsWithMember
+import java.net.URL
+import java.util.*
+
+/**
+ * BlockingClassLoader is a ClassLoader holding blacklist of packages and classes, as well as a parent ClassLoader.
+ * All classes, resources and packages contained in that List may not be used when attempting to load a Class or
+ * Resource.
+ *
+ * @property softMode
+ *          If activated, softMode will prevent an Exception to be thrown, if the Class is on the blacklist
+ * @property blacklist
+ *          the blacklist containing names of Classes, resources, as well as of packages, which may not be used when
+ *          loading a class or resource.
+ *
+ * @constructor
+ *          Instantiates a new BlacklistClassloader.
+ *
+ * @param parent
+ *          the parent ClassLoader
+ */
+class BlacklistClassloader(
```

Comment:
Should only implement a method that is used to decide if a class, package or resource is allowed to access.